pandas/core/arrays/base.py

```diff
@@ -333,6 +335,18 @@ def __iter__(self):
         for i in range(len(self)):
             yield self[i]
 
+    def __eq__(self, other: Any) -> Union[np.ndarray, "ExtensionArray"]:
+        """
+        Return for `self == other` (element-wise equality).
+        """
+        raise AbstractMethodError(self)
+
+    def __ne__(self, other: Any) -> Union[np.ndarray, "ExtensionArray"]:
```

Comment:
This should allow us to remove the equivalent `IntervalArray` definition here:

https://github.com/pandas-dev/pandas/blob/d149f4109de8610ccc4e3d229c037047b728a49e/pandas/core/arrays/interval.py#L609-L610

Reply:
`Union[np.ndarray, "ExtensionArray"]` -> `ArrayLike`

Reply:
@jschendel done!